test/test_ops_gradients.py

```diff
@@ -36,6 +36,13 @@ def test_fn_grad(self, device, dtype, op):
         else:
             self._grad_test_helper(device, dtype, op, op.get_op())
 
+    def setUp(self):
+        TestGradients.setUp(self)
+        # reset dynamo cache to avoid issues like
+        # https://github.com/pytorch/pytorch/issues/125967#issuecomment-2118483919
+        # which depends on test order.
+        torch._dynamo.reset()
+
```

Comment:
Btw, torch._dynamo.reset() might be expensive (when called so many times), so you might want to guard this on if we are doing PYTORCH_TEST_WITH_INDUCTOR